ls1intum/Artemis
# `Development`: Remove moved to the Lecture microservice features from the Artemis microservice

### Checklist
#### General


#### Server

### Motivation and Context


This PR is the final one for the migration of the Lecture microservice.

### Description

It removes the features related to lectures from Artemis which were moved to the Lecture microservice. Code review should be fast as there are only ~100 lines added and the deleted files already exist in the Lecture microsercvice /lecture.

Since code that was covered by tests is removed from Artemis there is a slight drop in the test coverage which required the change of the coverage rules.

### Steps for Testing

Prerequisites:
- Test server 6: [https://artemistest6.ase.in.tum.de/](https://artemistest6.ase.in.tum.de/) - it is already deployed there
- 1 Instructor (artemis_test_user_12 or you can create another instructor account using the server administration)
- 1 Student (artemis_test_user_1 or you can create another student account using the server administration)

1. Log in to Artemis on [https://artemistest6.ase.in.tum.de/](https://artemistest6.ase.in.tum.de/)
2. Navigate to Course Management
3. Open/Create a course and create a lecture => lecture should be successfully created
5. Edit the lecture => the lecture should be successfully updated
6. Add attachments and lecture units of all types to the lecture => should work fine
7. Navigate to course overview (as a strudent) and the lecture you created => the lecture should be there including the units and the attachments you added
8. Edit the lecture units => should work fine
9. Open the lecture and check that the exercise is removed from the lecture units list
10. Go back to Course Management as an instructor and delete lecture unit(s) => should work fine
11. Delete a lecture => the lecture should be successfully deleted

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|


### Screenshots
